src/test/lambda/wizards/samInitWizard.test.ts

```diff
@@ -3,605 +3,42 @@
-        it('uses user response as runtime', async function () {
-            const context: CreateNewSamAppWizardContext = new MockCreateNewSamAppWizardContext(
-                [],
-                Set<Runtime>(['nodejs10.x']),
-                ['npm'],
-                'myName',
-                [vscode.Uri.file(dir)],
-                Set<SamTemplate>([helloWorldTemplate]),
-                [],
-                [],
-                []
-            )
-            const wizard = new CreateNewSamAppWizard(context)
-            const args = await wizard.run()
-
-            assert.ok(args)
-            assert.strictEqual(args!.runtime, 'nodejs10.x')
-            assert.strictEqual(args!.dependencyManager, 'npm')
-        })
-
-        it('selects a runtime, restarts the step if a dependency manager is not set, and continues on', async function () {
-            const context: CreateNewSamAppWizardContext = new MockCreateNewSamAppWizardContext(
-                [],
-                Set<Runtime>(['java11', 'nodejs14.x']),
-                [undefined, 'npm'],
-                'myName',
-                [vscode.Uri.file(dir)],
-                Set<SamTemplate>([helloWorldTemplate]),
-                [],
-                [],
-                []
-            )
-            const wizard = new CreateNewSamAppWizard(context)
-            const args = await wizard.run()
+    let tester: WizardTester<CreateNewSamAppWizardForm>
 
-            assert.ok(args)
-            assert.strictEqual(args!.runtime, 'nodejs14.x')
-            assert.strictEqual(args!.dependencyManager, 'npm')
-        })
-
-        it('exits when cancelled', async function () {
-            const context: CreateNewSamAppWizardContext = new MockCreateNewSamAppWizardContext(
-                [],
-                Set<Runtime>(),
-                [],
-                'myName',
-                [vscode.Uri.file(dir)],
-                [],
-                [],
-                [],
-                []
-            )
-            const wizard = new CreateNewSamAppWizard(context)
-            const args = await wizard.run()
-
-            assert.ok(!args)
-        })
-
-        it('exits when a runtime is selected, a dependency manager is not, and then cancelled', async function () {
-            const context: CreateNewSamAppWizardContext = new MockCreateNewSamAppWizardContext(
-                [],
-                [Set<Runtime>('java11'), Set<Runtime>()],
-                [],
-                'myName',
-                [vscode.Uri.file(dir)],
-                [],
-                [],
-                [],
-                []
-            )
-            const wizard = new CreateNewSamAppWizard(context)
-            const args = await wizard.run()
-
-            assert.ok(!args)
-        })
+    beforeEach(function () {
+        tester = createWizardTester(new CreateNewSamAppWizard({ samCliVersion: '' } as any)) // the hard-cast is ok here since we only test control-flow
```

Comment:
The hard cast feels to me like a code smell. Specifically, you're hard casting to avoid the `regionProvider` and `awsContext` values. However, you're only using those values in the wizard for the [`createRegionPrompter` function],(https://github.com/aws/aws-toolkit-vscode/pull/2003/files#diff-e56abd315fb7d04fd2e48e51db13ac8f98de75e32e692d9f9b3fe35701357054R26), which I'm thinking is trying to do way too much for no apparent reason. Can you pull that functionality out elsewhere and just provide an array of regions in this constructor? That'll make the context easier to spoof and make that function easier to follow.

Reply:
Yeah I'll just break up the region stuff. I had a reason for piping the context everywhere at one point, but it's lost on me now.